tests/devices/qutrit_mixed/test_qutrit_mixed_apply_operation.py

```diff
@@ -0,0 +1,307 @@
+        assert qml.math.allclose(res, expected)
+
+    def test_batch_size_set_if_missing(self, ml_framework, one_qutrit_state):
+        """Tests that the batch_size is set on an operator if it was missing before."""
+        param = qml.math.asarray([0.1, 0.2], like=ml_framework)
+        state = one_qutrit_state
+        op = qml.TRX(param, 0)
+        op._batch_size = None  # pylint:disable=protected-access
+        state = apply_operation(op, state)
+        assert state.shape == (self.num_batched, 3, 3)
+        assert op.batch_size == self.num_batched
+
+
+@pytest.mark.parametrize("ml_framework", ml_frameworks_list)
+class TestChannels:  # pylint: disable=too-few-public-methods
+    """Tests that Channel operations are applied correctly."""
+
+    class CustomChannel(Channel):
+        num_params = 1
+        num_wires = 1
+
+        def __init__(self, p, wires, id=None):
+            super().__init__(p, wires=wires, id=id)
+
+        @staticmethod
+        def compute_kraus_matrices(p):
+            K0 = (np.sqrt(1 - p) * math.cast_like(np.eye(3), p)).astype(complex)
+            K1 = (
+                np.sqrt(p) * math.cast_like(np.array([[0, 0, 1], [1, 0, 0], [0, 1, 0]]), p)
+            ).astype(complex)
+            return [K0, K1]
+
+    def test_non_broadcasted_state(self, ml_framework, two_qutrit_state):
+        """Tests that Channel operations are applied correctly to a state."""
+        state = two_qutrit_state
+        test_channel = self.CustomChannel(0.3, wires=1)
+        res = apply_operation(test_channel, math.asarray(state, like=ml_framework))
+        flattened_state = state.reshape(9, 9)
+
+        mat = test_channel.kraus_matrices()
+
+        expanded_mats = [np.kron(np.eye(3), mat[i]) for i in range(len(mat))]
+        expected = np.zeros((9, 9)).astype(complex)
+        for i in range(len(mat)):
+            expanded_mat = expanded_mats[i]
+            adjoint_mat = np.conj(expanded_mat).T
+            expected += expanded_mat @ flattened_state @ adjoint_mat
+        expected = expected.reshape([3] * 4)
+
+        assert qml.math.get_interface(res) == ml_framework
+        assert res.shape == expected.shape
+        assert qml.math.allclose(res, expected)
+
+    def test_broadcasted_state(self, ml_framework, two_qutrit_batched_state):
+        """Tests that Channel operations are applied correctly to a batched state."""
+        state = two_qutrit_batched_state
+        num_batched = two_qutrit_batched_state.shape[0]
+
+        test_channel = self.CustomChannel(0.3, wires=1)
+        res = apply_operation(test_channel, math.asarray(state, like=ml_framework))
+
+        mat = test_channel.kraus_matrices()
+        expanded_mats = [np.kron(np.eye(3), mat[i]) for i in range(len(mat))]
+        expected = [np.zeros((9, 9)).astype(complex) for _ in range(num_batched)]
+        for i in range(num_batched):
+            flattened_state = state[i].reshape(9, 9)
+            for j in range(len(mat)):
```

Comment:
This same logic appears in quite a few places; recommend to abstract out as a function, and leverage `qml.matrix` as applicable.

Reply:
`qml.matrix` doesn't work for qutrit ops: https://github.com/PennyLaneAI/pennylane/issues/4367

Reply:
RIGHT... :cry:

Reply:
I did the best I could